### PR TITLE
Updated docs for Django releases, including details about fetching and processing translations.

### PR DESCRIPTION
**Pull request by nessita**

### CODE REVIEW

**Review by lhmzhou — APPROVED** [review]

File: docs/internals/howto-release-django.txt [hunk 1]
```diff
@@ -129,18 +142,44 @@ any time leading up to the actual release:
 #. Double-check that the release notes index has a link to the notes
    for the new release; this will be in ``docs/releases/index.txt``.
 
-#. If this is a feature release, ensure translations from Transifex have been
-   integrated. This is typically done by a separate translation's manager
-   rather than the releaser, but here are the steps. Provided you have an
-   account on Transifex:
+#. If this is a feature release (i.e. the final release of a new version),
```
Comment (felixxm) [inline docs/internals/howto-release-django.txt:145]: This is more confusing than clarifying anything. `4.0.2` is also the final release of a new version, but it's not what we want to mention here. I'd use `term`:
```suggestion
#. If this is a :term:`feature release`,
```

File: docs/internals/howto-release-django.txt [hunk 2]
```diff
@@ -129,18 +142,44 @@ any time leading up to the actual release:
...
+   ensure translations from Transifex have been integrated. This is typically
+   done by a separate translation's manager rather than the releaser, but here
+   are the steps. This process is a bit lengthy and cumbersome, for example,
```
Comment (felixxm) [inline docs/internals/howto-release-django.txt:148]: I'd see much value in complaining in docs about out own process :upside_down_face: 
```suggestion
   are the steps.
```

File: docs/internals/howto-release-django.txt [hunk 3]
```diff
@@ -129,18 +142,44 @@ any time leading up to the actual release:
...
+   All the new ``.po`` files should be manually and carefully inspected to
+   avoid committing a change in a file without any new translations. Also,
+   there shouldn't be any changes in the "plural forms": if there are any
+   (usually Spanish and French report changes for this) those will need
+   reverting.
+
+   Lastly, commit the changed/added files (both ``.po`` and ``.mo``) and create
+   a new PR targeting the stable branch of the corresponding release. See for
+   example this `PR updating translations for 4.2
+   <https://github.com/django/django/pull/16715>`_.
```
Comment (felixxm) [inline docs/internals/howto-release-django.txt:182]: I'd chop all "See..." 
```suggestion
   a new PR targeting the stable branch of the corresponding release
   (`example PR updating translations for 4.2
   <https://github.com/django/django/pull/16715>`_).
```

File: docs/internals/howto-release-django.txt [hunk 4]
```diff
@@ -447,9 +488,14 @@ You're almost done! All that's left to do now is:
...
-#. If it's a final release, update the current stable branch and remove the
-   pre-release branch in the `Django release process
-   <https://code.djangoproject.com/#Djangoreleaseprocess>`_ on Trac.
+#. If this was a final release:
+
+   #. Update the current stable branch and remove the pre-release branch in the
+      `Django release process
+      <https://code.djangoproject.com/#Djangoreleaseprocess>`_ on Trac.
+
+   #. Update djangoproject.com's download page (`see example PR
```
Comment (felixxm) [inline docs/internals/howto-release-django.txt:497]:
```suggestion
   #. Update djangoproject.com's download page (`example PR
```

File: docs/internals/howto-release-django.txt [hunk 5]
```diff
@@ -466,7 +512,8 @@ need to be done by the releaser.
 #. Create a new ``DocumentRelease`` object in the ``docs.djangoproject.com``
    database for the new version's docs, and update the
    ``docs/fixtures/doc_releases.json`` JSON fixture, so people without access
-   to the production DB can still run an up-to-date copy of the docs site.
+   to the production DB can still run an up-to-date copy of the docs site
+   (`example PR <https://github.com/django/djangoproject.com/pull/1446>`_).
```
Comment (felixxm) [inline docs/internals/howto-release-django.txt:516]:
```suggestion
   (`example PR <https://github.com/django/djangoproject.com/pull/1446>`__).
```

**Review by felixxm — COMMENTED** [review]
@nessita Thanks :+1: I left suggestions.

**Pull request merged**